sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReleaseInfo.java

```diff
@@ -24,65 +24,90 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.Properties;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import javax.annotation.Nullable;
 
 /**
  * Utilities for working with release information.
  */
 public final class ReleaseInfo extends GenericJson {
   private static final Logger LOG = LoggerFactory.getLogger(ReleaseInfo.class);
 
-  private static final String PROPERTIES_PATH =
-      "/org/apache/beam/sdk/sdk.properties";
+  private static final String MANIFEST_TIMESTAMP_ATTRIBUTE = "Build-timestamp";
 
   private static class LazyInit {
-    private static final ReleaseInfo INSTANCE =
-        new ReleaseInfo(PROPERTIES_PATH);
+    private static final ReleaseInfo INSTANCE = new ReleaseInfo();
   }
 
   /**
-   * Returns an instance of DataflowReleaseInfo.
+   * The string that will be used as a version if none can be ascertained.
+   */
+  public static final String UNKNOWN_VERSION_STRING = "Unknown";
+
+  /**
+   * Returns an instance of {@link ReleaseInfo}.
    */
   public static ReleaseInfo getReleaseInfo() {
     return LazyInit.INSTANCE;
   }
 
   @Key private String name = "Apache Beam SDK for Java";
-  @Key private String version = "Unknown";
+  @Key private String version = UNKNOWN_VERSION_STRING;
+  @Nullable @Key("build.date") private String timestamp = null;
```

Comment:
I don't feel strongly about keeping this name. I would be just as happy with `Build-timestamp` or whatever, straight from the manifest. But in case other code uses I didn't want to change it gratuitously.